File: paiza/ruby/C/C054.rb
# 標準入力から値を取得してinput_lineに入れる

def input
    gets.chomp.split.map!(&:to_i)
end

flag = false

ipt = input
N = ipt[0]
border = ipt[1]

t_x0 = input
t_i = t_x0[0]
x_i = t_x0[1]

(N - 1).times do |i|

  t_xi1 = input
  t_i1 = t_xi1[0]
  x_i1 = t_xi1[1]
   
  ave_v = (x_i1 - x_i).to_f / (t_i1 - t_i).to_f
   
  if ave_v > border
      flag = true
      break
  end
   
  t_i = t_i1
  x_i = x_i1
end

puts flag == true ? 'YES' : 'NO'

File: paiza/javascript/C054.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

var ipt = []; // 標準入力から与えられた各行を保存する変数：input_lines
var reader = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});
reader.on('line', (line) => {
  ipt.push(line);
});
reader.on('close', () => {
    const input = ipt[0].split(' ');
    const N = parseInt(input[0]);
    const border = parseInt(input[1]);
    let t = [];
    let x = [];
    let flag = true;
    
    for (let i = 1; i < N + 1; i++){
 
        let params = ipt[i].split(' ');
        t.push(parseInt(params[0]));
        x.push(parseInt(params[1]));
    }

    for (let j = 0; j < N - 1; j++){
        dt = t[j + 1] - t[j];
        dx = x[j + 1] - x[j];

        if ((dx / dt) > border){
            flag = false;
            break;
        }
    }
    
    console.log((flag === true) ? 'NO' : 'YES');
});